tests/test_unittest/test_backend_api.py

```diff
@@ -28,30 +28,44 @@ def init_1(input_field: Field[float]):  # type: ignore
         input_field = 1  # noqa - unused var is in/out field
 
 
-def test_generate_computation(backend):
+def test_generate_computation(backend, tmp_path):
     """
-    Test the :py:meth:`gt4py.backend.Backend.generate_computation` method.
-
-    Assumption:
-
-        * All registered backends support the `generate_computation` api.
-
-    Actions:
-
-        Generate the computation source code for a simple stencil.
-
-    Outcome:
-
-        The computation source code and file hierarchy specification is returned.
-
+    Test the :py:meth:`gt4py.backend.CLIBackendMixin.generate_computation` method.
```

Comment:
```suggestion
    Test if the :py:meth:`gt4py.backend.CLIBackendMixin.generate_computation` generates code.
```
I'm not convinced by this kind of units tests. Testing for `init_1_src` being set is fine, but testing for `computation.cpp` and `computation.cu` and in particular for not having `bindings.cpp` seems overly specific. After all if someone changes this he has to do it everywhere and the regression tests will not only catch this, but also point the location where he has to. I would just simplify this, if `c++` is the language I expect `cpp` files, cuda `cu``and `cpp` I guess and python `py`.

Reply:
I don't like it much either, however this is actually very unlikely to change and if someone does, then they should change it in a way this can be tested. And hopefully they will move from hardcoding those names to something better.

If I remove these tests I can conversely not make changes for the CLI in the near future while guaranteeing the same behaviour.